price-feeder/oracle/provider/coinbase.go

```diff
@@ -480,7 +481,10 @@ func (p *CoinbaseProvider) resetReconnectTimer() {
 // 3. Expect a 'pong' as a response. If the response message is not received within
 // N seconds, please raise an error or reconnect.
 func (p *CoinbaseProvider) reconnect() error {
-	p.wsClient.Close()
+	err := p.wsClient.Close()
+	if err != nil {
+		return sdkerrors.Wrap(types.ErrProviderConnection, fmt.Errorf("error closing Coinbase websocket %w", err))
```

Comment:
1. You can use a shortcut: `types.ErrProviderConnection.Wrapf("error closing Coinbase websocket %v", err)`. Note, here we use `%v`, instead of `%w`, because `Wrapf` is using `fmt.Sprintf` under the hood.
2. I have always a dilemma if we should 
    * wrap a source error (`err`) on top of our base error (this way we loose `err` structure and  `ErrProviderConnection` is the causer (parent error) - this is the way how the SDK operates.
    * or to wrap the connection error on top of a message: (the alternative I've presented earlier): 
    
        ```go
        func connectionErr(provider string) error { 
          return fmt.Errorf("error closing %s websocket: %w", provider,  err)
        }
        ```

Since here we are returning a generic connection error, then I'm more inclined with the first approach. So... if you agree, then please use the shourtcut.

Reply:
Resolved in a29979fd52212c4c2f214c3cfb84eaccb9aac414